docs/ko/faq/how-do-i-use-kotlin-value-class.md

```diff
@@ -0,0 +1,129 @@
+# Kotlin value class 를 사용하려면 어떻게 해야할까요?
+
+엔티티의 프로퍼티를 kotlin의 [`value class`](https://kotlinlang.org/docs/inline-classes.html)로 선언할 수 있습니다.
+
+```kotlin
+@Entity
+class User(
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    val id: UserId = UserId(0),
+)
+
+@JvmInline
+value class UserId(private val value: Long)
+
+@Service
+class UserService(
+    private val jpqlRenderContext: JpqlRenderContext,
+    private val entityManager: EntityManager,
+) {
+
+    fun findById(userId: UserId): User? {
+        val query = jpql {
+            select(
+                entity(User::class)
+            ).from(
+                entity(User::class),
+            ).where(
+                path(User::id).equal(userId)
+            )
+        }
+
+        return entityManager.createQuery(query, jpqlRenderContext).apply { maxResults = 1 }.resultList.firstOrNull()
+    }
+}
+```
+
+하지만 추가적인 설정 없이 Hibernate를 사용해 Kotlin JDSL을 통해 조회하면 에러가 발생합니다.
+
+```
+org.hibernate.type.descriptor.java.CoercionException: Cannot coerce value 'UserId(value=1)' [com.example.entity.UserId] to Long
+...
+```
+
+이를 해결하려면 Kotlin JDSL이 매개 변수로 전달되는 `value class`의 unboxing이 필요합니다.
+unboxing은 다음 방안 중 하나를 선택해서 수행할 수 있습니다.
+
+### JpqlValue용 커스텀 JpqlSerializer
+
+에러를 해결하기 위해 `EntityManager`에 인자들을 `value class` 그 자체로 넘기지 않고 unboxing한 값을 넘겨야합니다.
+Kotlin JDSL은 `JpqlValueSerializer` 클래스에서 인자들을 추출하는 역할을 담당합니다.
+따라서 기본 제공하는 클래스 대신 커스텀 Seriailzer를 등록해야 합니다.
+
+먼저 다음과 같은 커스텀 Seriailzer를 생성합니다.
+
+```kotlin
+class ValueClassAwareJpqlValueSerializer(
```

Comment:
@shown How about adding this code to the actual jdsl rather than the documentation? It seems inevitable to use reflection. From the perspective of using value class, there seems to be enough merit.

Reply:
The reason for not providing value classes as built-in in the Kotlin JDSL is not only because it adds points to use reflection. It's because we don't know how value classes will change in the future. Please see the following comment for more details. 

https://github.com/line/kotlin-jdsl/issues/559#issuecomment-1880995684

Reply:
It seems ambiguous. I think it would be a good idea to consider it again in the future. Consider reapplying when uncertainty is removed!

Reply:
But if that's the case, wouldn't it be a good idea to also include the reasons you mentioned above in the document above? Since there are still some ambiguous aspects to support, why not just leave a guide as a document?

Reply:
Once the Valhalla project in Java is finished and Kotlin supports multiple values in a value class instead of just one, I'll consider supporting it. 

And I think it's better not to document this background for now, because I think it would be better to interview users and see what questions they have after reading this guide, and then write it up in a document.

Reply:
All right. I respect your opinion.